### PR TITLE
Feat/alchemy api key for ethereum

### PR DESCRIPTION
**Pull request by dlipicar**
### What does the PR do

status-go part: https://github.com/status-im/status-go/pull/4089

Closes https://github.com/status-im/status-desktop/issues/12312

Use the new Alchemy API keys for Ethereum Mainnet and Goerli

### CODE REVIEW

File: src/app_service/common/network_constants.nim [hunk 1]
```diff
@@ -32,6 +32,24 @@ let INFURA_TOKEN_SECRET_RESOLVED* =
...
     INFURA_TOKEN_SECRET
 
+# set via `nim c` param `-d:ALCHEMY_ETHEREUM_MAINNET_TOKEN:[token]`; should be set in CI/release builds
+const ALCHEMY_ETHEREUM_MAINNET_TOKEN {.strdefine.} = ""
+let ALCHEMY_ETHEREUM_MAINNET_TOKEN_ENV = $getEnv("ALCHEMY_ETHEREUM_MAINNET_TOKEN")
+let ALCHEMY_ETHEREUM_MAINNET_TOKEN_RESOLVED* =
+  if ALCHEMY_ETHEREUM_MAINNET_TOKEN_ENV != "":
+    ALCHEMY_ETHEREUM_MAINNET_TOKEN_ENV
+  else:
+    ALCHEMY_ETHEREUM_MAINNET_TOKEN
```
Comment (jakubgs) [inline src/app_service/common/network_constants.nim:42]: Do we really need to repeat the same thing like 8 or more times? Could we not make a function for this?

Comment (dlipicar) [reply src/app_service/common/network_constants.nim:42]: can do

Comment (dlipicar) [reply src/app_service/common/network_constants.nim]: done

**Review by jakubgs — APPROVED** [review]
Much nicer. Ping me when you're happy with it so I can tag the library release.

**Review by alaibe — APPROVED** [review]

**Comment by jakubgs** [comment]
I have merged the `status-jenkins-lib` PR and tagged under `v1.7.16`:

* https://github.com/status-im/status-jenkins-lib/pull/75

**Pull request merged**